common/src/main/java/com/android/designcompose/common/GenericDocContent.kt

```diff
@@ -43,18 +43,26 @@ class GenericDocContent(
     fun save(filepath: File, feedback: FeedbackImpl) {
         feedback.documentSaveTo(filepath.absolutePath, docId)
         try {
-            val serializer = BincodeSerializer()
-            header.serialize(serializer)
-            document.serialize(serializer)
+            val bytes = toSerializedBytes(feedback)
             val output = FileOutputStream(filepath)
-            output.write(serializer._bytes)
-            output.write(imageSessionData)
+            output.write(bytes)
             output.close()
             feedback.documentSaveSuccess(docId)
         } catch (error: Throwable) {
             feedback.documentSaveError(error.toString(), docId)
         }
     }
+    fun toSerializedBytes(feedback: FeedbackImpl): ByteArray? {
+        try {
+            val serializer = BincodeSerializer()
+            header.serialize(serializer)
+            document.serialize(serializer)
+            return serializer._bytes.toUByteArray().toByteArray() + imageSessionData
```

Comment:
Can a copy be avoided if you were to use asByteArray instead of toByteArray?

Reply:
Oh yea, good point.

Reply:
Rob, just checking the use of toByteArray - would asByteArray be better or does it not work in this case?

Reply:
Yes it should work and I remember changing it, but it somehow didn't make it. Hmm, I'll make a follow up pull request.